modules/kubernetes-addons/karpenter/locals.tf

```diff
@@ -26,48 +26,34 @@ locals {
   ]
 
   default_helm_config = {
-    name                       = "karpenter"
-    chart                      = "karpenter"
-    repository                 = "https://charts.karpenter.sh"
-    version                    = "0.6.3"
-    namespace                  = local.namespace
-    timeout                    = "300"
-    create_namespace           = false
-    values                     = local.default_helm_values
-    set                        = []
-    set_sensitive              = null
-    lint                       = true
-    wait                       = true
-    wait_for_jobs              = false
-    description                = "karpenter Helm Chart for Node Autoscaling"
-    verify                     = false
-    keyring                    = ""
-    repository_key_file        = ""
-    repository_cert_file       = ""
-    repository_ca_file         = ""
-    repository_username        = ""
-    repository_password        = ""
-    disable_webhooks           = false
-    reuse_values               = false
-    reset_values               = false
-    force_update               = false
-    recreate_pods              = false
-    cleanup_on_fail            = false
-    max_history                = 0
-    atomic                     = false
-    skip_crds                  = false
-    render_subchart_notes      = true
-    disable_openapi_validation = false
-    dependency_update          = false
-    replace                    = false
-    postrender                 = ""
+    name        = local.name
+    chart       = local.name
+    repository  = "https://charts.karpenter.sh"
+    version     = "0.6.3"
+    namespace   = local.name
+    timeout     = "300"
+    values      = local.default_helm_values
+    set         = []
+    description = "karpenter Helm Chart for Node Autoscaling"
   }
 
   helm_config = merge(
     local.default_helm_config,
     var.helm_config
   )
 
+  irsa_config = {
+    kubernetes_namespace              = local.helm_config["namespace"]
+    kubernetes_service_account        = local.service_account_name
+    create_kubernetes_namespace       = true
+    create_kubernetes_service_account = true
+    iam_role_path                     = "/"
+    eks_cluster_id                    = var.addon_context.eks_cluster_id
```

Comment:
Remove `eks_cluster_id` and `tags` as they are now available in addon_context which is passed to the helm_addon.

Reply:
updated